### DISCUSSION TITLE
What's the best way to make walls/platforms

### DISCUSSION THREAD

**Discussion by ufukbakan** [excaliburjs/Excalibur#2671] [Q&A]
Are walls/platforms actors too?

If yes then assume that I've 32x32 single wall sprite, but i want to create a wall platform 320x32 which needs its sprite to be repeated 10x in horizontal direction. How can i do that?

I read tilemap documentation but it really is not helpfull it only shows constructor arguments it doesnt tell how to bind it to an actor or to scene?

**Comment by eonarheim** [comment]
HI @ufukbakan !

Good question, you can use both Actors/TileMaps depending on your preference and needs. I'll update the documentation with more examples of usage!

`TileMap`s can have the graphics set on individual tiles if you know the x/y. You might use these if you are needing to render something that will be oriented in a uniform grid. Also TileMaps are a more efficient from Excalibur's perpective since Tiles are more lightweight. 

```typescript

const tileMap = new ex.TileMap({
  name: 'tilemap',
  pos: ex.vec(0, 0),
  tileWidth: 32,
  tileHeight: 32,
  rows: 10,
  columns: 100
});
for(let tile of tileMap.tiles) {
  if (tile.y > 5) {
    // floor that is solid with a graphic!
    tile.solid = true;
    tile.addGraphic(blocks);
  }
}
```
If you don't want to build TileMaps by hand you could look into the excalibur tiled plugin to load external maps https://github.com/excaliburjs/excalibur-tiled


Another strategy using Actors: you could create 10 320x32 Actors and place them horizontally. You might use actors if you aren't going to be aligned to a grid, or you have features that need to move around independently. 

Somthing like this:

```typescript
for (let i = 0; i < 10; i++) {
  const wall = new ex.Actor({
    x: i * 32,
    Y: 300,
    width: 32,
    height: 32,
  })
  wall.graphics.use(blocks);
  game.add(wall);
}

```

**Reply by ufukbakan** [reply]
How to place this tilemap in game via an actor? I still dont get it i cant graphics.use(tilemap) or graphics.use(tilemap.getTile(x,y))

**Reply by eonarheim** [reply]
Hi @ufukbakan 

A `TileMap` is like an actor itself, you can place it by setting it's position directly or you could parent it to an Actor.

https://excaliburjs.com/docs/tilemap/#tilemap

Example of constructing a tilemap and setting position
```typescript
const game = new ex.Engine({...});
const tileMap = new ex.TileMap({
  pos: ex.vec(50, 50)
});
// or
tileMap.pos = ex.vec(50, 50);

// Tilemap is now drawn
game.add(tileMap);
```

Example of parenting a tilemap to an actor
```typescript
const game = new ex.Engine({...});

const actor = new ex.Actor({...});
actor.pos = ex.vec(50, 50);
game.add(actor);


// TileMap will be drawn relative to the parent actor position
const tileMap = new ex.TileMap({...});
actor.addChild(tileMap);
```
